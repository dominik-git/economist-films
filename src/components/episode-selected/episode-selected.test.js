// @flow
import React from 'react';
import renderer from 'react-test-renderer';
import { mount, shallow } from 'enzyme';
import EpisodeSelected from './episode-selected';

jest.mock('./parts/episode-description', () =>
  jest.fn(() => <div>EpisodeDescription</div>),
);

function connectEvent(event, type, wrapper) {
  const changedEvent: Object = event;
  changedEvent.key = type;
  const episodeSelected: Object = wrapper.instance();
  episodeSelected.handleKeyPress(event);
}

describe('HomeContainer: ', () => {
  test('renders with data', () => {
    const tree : string = renderer.create(<EpisodeSelected
      id={3}
      url="url xyz"
      title="title xyz"
      subtitle="subtitle xyz"
      description="description xyz"
      closePopupFunction={() => {}}
    />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('handles keyboard events', () => {
    let episodeSelected = mount(<EpisodeSelected
      id={3}
      url="url xyz"
      title="title xyz"
      subtitle="subtitle xyz"
      description="description xyz"
      closePopupFunction={() => {}}
    />);
    const event = new Event('keyDown');
    expect(episodeSelected.state().selectedItem).toEqual(0);
    connectEvent(event, 'ArrowRight', episodeSelected);
    expect(episodeSelected.state().selectedItem).toEqual(1);
    connectEvent(event, 'ArrowRight', episodeSelected);
    expect(episodeSelected.state().selectedItem).toEqual(2);
    connectEvent(event, 'ArrowRight', episodeSelected);
    expect(episodeSelected.state().selectedItem).toEqual(2);
    connectEvent(event, 'ArrowLeft', episodeSelected);
    expect(episodeSelected.state().selectedItem).toEqual(1);
    connectEvent(event, 'Enter', episodeSelected);
    connectEvent(event, 'ArrowLeft', episodeSelected);
    expect(episodeSelected.state().selectedItem).toEqual(0);
    connectEvent(event, 'ArrowLeft', episodeSelected);
    expect(episodeSelected.state().selectedItem).toEqual(0);
    connectEvent(event, 'ArrowUp', episodeSelected);
    connectEvent(event, 'Backspace', episodeSelected);
    connectEvent(event, 'Shift', episodeSelected);
    episodeSelected = mount(<EpisodeSelected
      id={3}
      url="url xyz"
      title="title xyz"
      subtitle="subtitle xyz"
      description="description xyz"
      closePopupFunction={null}
    />);
    connectEvent(event, 'ArrowUp', episodeSelected);
    connectEvent(event, 'Backspace', episodeSelected);
  });
  test('mounts/unmounts', () => {
    const wrapper = shallow(<EpisodeSelected
      id={3}
      url="url xyz"
      title="title xyz"
      subtitle="subtitle xyz"
      description="description xyz"
      closePopupFunction={() => {}}
    />);
    wrapper.unmount();
  });
});
