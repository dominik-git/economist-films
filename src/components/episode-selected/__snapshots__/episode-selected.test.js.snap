// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeContainer:  renders with data 1`] = `
<div
  className="episode-selected"
>
  <div
    className="episode-selected__image-wrapper"
  >
    <img
      alt="title xyz"
      className="episode-selected__image episode-selected__image--selected"
      src="url xyz"
    />
  </div>
  <div
    className="episode-selected__info-container"
  >
    <div>
      EpisodeDescription
    </div>
    <div
      className="episode-buttons"
    >
      <a
        className="episode-buttons__watch"
        onClick={[Function]}
        style={Object {}}
      >
        Watch Now
      </a>
      <a
        className="episode-buttons__learn"
        onClick={[Function]}
        style={Object {}}
      >
        Learn More
      </a>
    </div>
  </div>
</div>
`;
